### ISSUE TITLE
hwmon config file for OCC, VDD, Ambient

### ISSUE THREAD

**Issue by rfrandse** [openbmc/openbmc#1183]
Updated description: Add phosphor-hwmon config files for the OCC, VDD, Ambient.

Old description:
```
This Story falls under "Get Temperature sensors working" :: 

- Other temps come from occ
- i.	OCC active - How do we find out that the OCC is active?
- v.	Config file for the OCC sensors to feed into hwmon (should come as MRW update) - 1pt
- vi.	Some mapping mechanism from the hwmon sensor to what it’s for (i.e. dimm, cpu, …) - 1pt
```

**Comment by bradbishop** [comment]
Use this story to add phospor-hwmon config files for the OCC.

**Comment by spinler** [comment]
Current open question:
- what path do I use for the config file under /etc/default/obmc/hwmon?  I2C hwmon files look like:  `./ahb/apb/i2c@1e78a000/i2c-bus@400/tmp275@4a.conf`

**Comment by bradbishop** [comment]
Had a meeting with @jk and @eddiejames tonight about this.  The direction coming out of that was that we'll use the same setup as the existing P8 OCC driver.  The general idea is that the IPMI sensor number will go in the label attribute:

temp1_label = 121

The config files will have to map this number to a name like core0.

One open question I have is whether or not the index will be stable.  As in, for a given system, can userspace assume that temp1 will always have the same label?  If userspace cannot..we'll have to implement a level of indirection to go from the label -> input in phosphor-hwmon that we currently don't have.

**Comment by bradbishop** [comment]
@eddiejames when you know what its going to be, can you post in here what the the paths will look like under /sys/devices and /sys/firmware?

**Comment by jk** [comment]
I'm pretty sure I had no meeting with @bradbishop.

**Comment by bradbishop** [comment]
Need to know what of_node and device in /sys/class/hwmon/hwmonX will point at for occ devices....
@eddiejames ^

**Comment by eddiejames** [comment]
/sys/firmware/devicetree/base/gpio-fsi/cfam@0,0/sbefifo@2400/occ@1/occ-hwmon@1
/sys/devices/platform/gpio-fsi/fsi0/slave\@00\:00/00\:00\:00\:06/sbefifo1-dev0/occ1-dev0/hwmon/hwmon6/

for second proc:
/sys/firmware/devicetree/base/gpio-fsi/cfam@1,0/sbefifo@2400/occ@2/occ-hwmon@2
/sys/devices/platform/gpio-fsi/fsi0/slave\@00\:00/00\:00\:00\:0a/fsi1/slave\@01\:00/01\:00\:00\:06/sbefifo1-dev0/occ1-dev0/hwmon/hwmon6/

**Comment by geissonator** [comment]
Questions and answers from todays meeting
- What specifically does this track?  Just OCC sensors or VDD and Ambient?
-- It tracks all of them
-- VDD is for processors (ir chip, cbostic getting into device tree)
-- ambient is in iio, Brandon has story to make this work (bmp280 (non ga sensor)) DTS310 is coming (will need a different config) - cards coming middle of May
-- Put conf file in for DTS310, wont be testable but we'll have it and can tweak if needed when hw shows up
- Can't guarantee instance numbers for occ sensors?
-- #1633 is where this gets addressed, review it and see if more questions

**Comment by geissonator** [comment]
Task List:
- [x] Add in DPS310 config file - https://gerrit.openbmc-project.xyz/#/c/4338/
- [ ] Add in IR35221 config file (dependency on #1177 - no hwmon yet)
- [ ] Add in OCC config file (this will require integration with #1633) 
- [ ] Verify all required sensors are available via dbus objects

**Comment by rfrandse** [comment]
https://gerrit.openbmc-project.xyz/4440 Add in OCC hwmon config file
Resolves #1183 hwmon config file for OCC, VDD, Ambient
